### PR TITLE
[SPRING] Generate constants for paths (#1385)

### PR DESCRIPTION
**Pull request by adrianiacobghiula**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.4.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@diyfr
@bbdouglas (2017/07) @JFCote (2017/08) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01)
@wing328 

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by dragosnutu** [comment]
Nice addition, I already did it it 'kind of' with using custom templates. But we've generated the paths in the same api interface. Will be this available in 4.0.0 release ?

File: modules/openapi-generator/src/main/resources/JavaSpring/apiRoute.mustache [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+* NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech) ({{{generatorVersion}}}).
+* https://openapi-generator.tech
+* Do not edit the class manually.
+*/
+package {{package}};
+
+{{#operations}}
+public interface {{classname}}Route {
```
Comment (cbornet) [inline modules/openapi-generator/src/main/resources/JavaSpring/apiRoute.mustache]: Constants should not be defined in interfaces : https://rules.sonarsource.com/java/tag/bad-practice/RSPEC-1214

**Comment by cbornet** [comment]
What is the gain of this ? I find it easier to read URLs directly in controllers. But that may be just me...

**Comment by wing328** [comment]
Closing this as there's no update

@adrianiacobghiula let us know if you still want to proceed with this PR.

**Pull request closed without merging**